lib/lcset.gi

```diff
@@ -0,0 +1,139 @@
+##############################################################################
+##
+#W  lcset.gi                 GAP4 package `Utils'                Chris Wensley
+##
+#Y  Copyright (C) 2015-2022, The GAP Group 
+
+#############################################################################
+##
+#O  LeftCoset( <g>, <U> ) . . . . . . . . forms a left coset of U by g \in G 
+##
+InstallMethod( LeftCoset, "generic method for left cosets", true, 
+    [ IsObject, IsGroup ], 0,
+function( g, U )
+    local lc;
+
+    lc := rec();
+    ObjectifyWithAttributes( lc, IsLeftCosetType,
+        Representative, g,
+        ActingDomain, U, 
```

Comment:
Strictly speaking, this is not correct, as left cosets are not external sets, and `ActingDomain` is defined only for external sets. But I guess this is a fair "abuse of notation" here.

Reply:
Thinking some more of it, of course you could still view it as an external set (that's what you declared in the definition of `IsLeftCosetObj`, too!) by simply taking the action  $(hU, g) \mapsto g^{-1}hU$, so I guess you are good to go after all. But to be a proper external set, a few more things need to be defined, I think... Indeed, this is what `RightCoset` does:
```gap
  fam:=FamilyObj(U);
  if not IsBound(fam!.rightCosetsDefaultType) then
    fam!.rightCosetsDefaultType:=NewType(fam,IsRightCosetDefaultRep and
          HasActingDomain and HasFunctionAction and HasRepresentative
          and HasCanonicalRepresentativeDeterminatorOfExternalSet);
  fi;

  d:=rec();
  ObjectifyWithAttributes(d,fam!.rightCosetsDefaultType,
    ActingDomain,U,FunctionAction,OnLeftInverse,Representative,g,
    CanonicalRepresentativeDeterminatorOfExternalSet,
    RightCosetCanonicalRepresentativeDeterminator);
```